src/App.tsx

```diff
@@ -7,33 +9,64 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = this.setTimer;
+    document.addEventListener('click', this.handlerClock);
+    document.addEventListener('contextmenu', this.handlerClock);
+  }
+
+  componentDidUpdate(_: Readonly<{}>, prevState: Readonly<State>) {
+    const prevName = prevState.clockName;
+    const currentName = this.state.clockName;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+    if (prevName !== currentName) {
+      console.log(`Renamed from <${prevName}> to <${currentName}>`); // eslint-disable-line
+    }
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  handlerClock = (event: MouseEvent) => {
+    if (event.type === 'contextmenu') {
+      event.preventDefault();
+      this.setState({ hasClock: false });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+      clearInterval(this.setTimer);
+    }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    if (event.type === 'click') {
+      this.setState({ hasClock: true });
 
-        {' time is '}
+      this.setTimer = window.setInterval(this.handlerSetInterval, 3300);
```

Comment:
It's overcomplicated. You don't need to clear and create new timer for calling getRandomName © Vova